chaco/plot_containers.py

```diff
@@ -136,145 +137,12 @@ def get_preferred_size(self, components=None):
             self._cached_preferred_size = self.outer_bounds[:]
             return self.outer_bounds
 
-        if components is None:
-            components = self.components
-
-        ndx = self.stack_index
-        other_ndx = 1 - ndx
-
-        no_visible_components = True
-        total_size = 0
-        max_other_size = 0
-        for component in components:
-            if not self._should_layout(component):
-                continue
-
-            no_visible_components = False
-
-            pref_size = component.get_preferred_size()
-            total_size += pref_size[ndx] + self.spacing
-            if pref_size[other_ndx] > max_other_size:
-                max_other_size = pref_size[other_ndx]
-
-        if total_size >= self.spacing:
-            total_size -= self.spacing
-
-        if (self.stack_dimension not in self.resizable) and (
-            self.stack_dimension not in self.fit_components
-        ):
-            total_size = self.bounds[ndx]
-        elif no_visible_components or (total_size == 0):
-            total_size = self.default_size[ndx]
-
-        if (self.other_dimension not in self.resizable) and (
-            self.other_dimension not in self.fit_components
-        ):
-            max_other_size = self.bounds[other_ndx]
-        elif no_visible_components or (max_other_size == 0):
-            max_other_size = self.default_size[other_ndx]
-
-        if ndx == 0:
-            self._cached_preferred_size = (
-                total_size + self.hpadding,
-                max_other_size + self.vpadding,
-            )
-        else:
-            self._cached_preferred_size = (
-                max_other_size + self.hpadding,
-                total_size + self.vpadding,
-            )
-
-        return self._cached_preferred_size
+        return stacked_preferred_size(self, components=components)
```

Comment:
Note that there is some duplication here.

`stacked_preferred_size` does not handle the case where `resizable=""` - and we might want to update the underlying enable util to handle this case - but until then, we will have to explicitly handle the `if self.fixed_preferred_size is not None:` case AND the `if self.resizable == "":` case - and implicitly handle the `if self.fixed_preferred_size is not None:` case again inside the enable util, which will not do anything. under the circumstances.